### PR TITLE
refactor(040): F6 lote 6.3 — triagem/tipagem testes core dose/schemas (repositories: 350→85)

### PR DESCRIPTION
**Pull request by coelhotv**
## Escopo
Lote 6.3 do épico 040 — queima de dívida de testes nível A do core (§F6 do playbook). Rubrica MATA/TIPA aplicada: nenhum teste caiu em MATA (mock-contra-mock, snapshot morto, implementação frágil) — todos exercitam lógica real de repository via mock builder Supabase fluente. Todos os erros TS eram `implicitly has an 'any' type` na infra de mock (não bugs de lógica), então classificados TIPA e corrigidos com anotações `: any` diretas nos helpers `makeBuilder`/`makeClient`/`makeQueueClient` e nos binding elements de `.find/.filter/.map`.

## Arquivos (11 arquivos de repositories/__tests__/)
createDoseInstanceRepository{,.countByStatus,.missedSweep}.test.ts, createFeedbackRepository, createMedicineRepository, createProfileRepository, createProtocolRepository{,.lifecycle}, createPurchaseRepository, createStockRepository, createTreatmentPlanRepository.

## Fixes pontuais (além de `: any` mecânico)
- `createDoseInstanceRepository.test.ts`: `findAnchorInstance` retorna `DoseInstance | null` — `out.id` → `out?.id`; teste de `protocolId: null` (edge-case proposital) recebeu `null as any`.
- `createStockRepository.test.ts`: `decreaseStock`/`adjustToBalance` testados com `null` proposital no 3º arg (API pública exige `string`) — `null as any` no call-site do teste, assinatura pública NÃO alterada.
- `createProtocolRepository.test.ts`: `.find()` pode retornar `undefined` (TS18048) — `!` non-null assertion, pois o teste já assume presença via `.toMatchObject`.

## Contadores
`MAX_A_TESTS_CORE`: 350 → **85** (alvo do lote era ≤120 — batido com folga; ratchet abaixado no mesmo commit).

## Testes MATA
Nenhum — todos os 350 erros eram infra de mock sem tipo, não testes falsos/frágeis.

## Triagem pendente
Nenhuma — os 21 arquivos de teste do bucket `testes A core` foram todos inspecionados; os 85 erros residuais (~85, nos 10 arquivos de `schemas/__tests__/` + `services/__tests__/anvisaDatabase.test.ts`) não foram tocados neste lote (fora do escopo de `repositories/`) e ficam disponíveis pro próximo lote ou pro residual (já abaixo do teto ≤120).

## Testes
`npx vitest run src/repositories/__tests__` (packages/core): 167/167 verdes. `strict-island.sh`: verde com teto novo.

Parar aqui — merge do lote na fase-6 é fora da sessão.

## AI Review Response

| Suggestion | Priority | Decision | Reason |
|------------|----------|----------|--------|
| createMedicineRepository.test.ts:241 `.find()` sem optional chaining (insert) | Medium | Applied | Commit 08a3c9cd |
| createMedicineRepository.test.ts:255 `.find()` sem optional chaining (update) | Medium | Applied | Commit 08a3c9cd |
| createTreatmentPlanRepository.test.ts:212 `.find()` sem optional chaining (insert) | Medium | Applied | Commit 08a3c9cd |
| createTreatmentPlanRepository.test.ts:225 `.find()` sem optional chaining (update) | Medium | Applied | Commit 08a3c9cd |
| createProfileRepository.test.ts:79 `.find()` sem optional chaining (eq) | Medium | Applied | Commit 08a3c9cd |

### CODE REVIEW

File: packages/core/src/repositories/__tests__/createMedicineRepository.test.ts [hunk 1]
```diff
@@ -237,8 +237,8 @@ describe('createMedicineRepository — parity', () => {
       await repoWeb.create(VALID_MEDICINE)
       await repoMob.create(VALID_MEDICINE)
 
-      const webInsert = clientWeb._builder._calls.find(([m]) => m === 'insert')[1][0][0]
-      const mobInsert = clientMob._builder._calls.find(([m]) => m === 'insert')[1][0][0]
+      const webInsert = clientWeb._builder._calls.find(([m]: any) => m === 'insert')[1][0][0]
+      const mobInsert = clientMob._builder._calls.find(([m]: any) => m === 'insert')[1][0][0]
```
[the remark this answers is not in the document]
Comment (coelhotv) [reply packages/core/src/repositories/__tests__/createMedicineRepository.test.ts]: @gemini-code-assist fix applied in commit 08a3c9cd — can you validate?

File: packages/core/src/repositories/__tests__/createMedicineRepository.test.ts [hunk 2]
```diff
@@ -251,8 +251,8 @@ describe('createMedicineRepository — parity', () => {
       await repoWeb.update('id-1', { name: 'Atualizado' })
       await repoMob.update('id-1', { name: 'Atualizado' })
 
-      const webUpdate = clientWeb._builder._calls.find(([m]) => m === 'update')[1][0]
-      const mobUpdate = clientMob._builder._calls.find(([m]) => m === 'update')[1][0]
+      const webUpdate = clientWeb._builder._calls.find(([m]: any) => m === 'update')[1][0]
+      const mobUpdate = clientMob._builder._calls.find(([m]: any) => m === 'update')[1][0]
```
[the remark this answers is not in the document]
Comment (coelhotv) [reply packages/core/src/repositories/__tests__/createMedicineRepository.test.ts]: @gemini-code-assist fix applied in commit 08a3c9cd — can you validate?

File: packages/core/src/repositories/__tests__/createTreatmentPlanRepository.test.ts [hunk 3]
```diff
@@ -207,8 +207,8 @@ describe('createTreatmentPlanRepository — parity', () => {
       await repoWeb.create(VALID_PLAN)
       await repoMob.create(VALID_PLAN)
 
-      const webInsert = clientWeb._builder._calls.find(([m]) => m === 'insert')[1][0][0]
-      const mobInsert = clientMob._builder._calls.find(([m]) => m === 'insert')[1][0][0]
+      const webInsert = clientWeb._builder._calls.find(([m]: any) => m === 'insert')[1][0][0]
+      const mobInsert = clientMob._builder._calls.find(([m]: any) => m === 'insert')[1][0][0]
       expect(webInsert).toEqual(mobInsert)
```
[the remark this answers is not in the document]
Comment (coelhotv) [reply packages/core/src/repositories/__tests__/createTreatmentPlanRepository.test.ts]: @gemini-code-assist fix applied in commit 08a3c9cd — can you validate?

File: packages/core/src/repositories/__tests__/createTreatmentPlanRepository.test.ts [hunk 4]
```diff
@@ -221,8 +221,8 @@ describe('createTreatmentPlanRepository — parity', () => {
       await repoWeb.update('id-1', { name: 'Atualizado' })
       await repoMob.update('id-1', { name: 'Atualizado' })
 
-      const webUpdate = clientWeb._builder._calls.find(([m]) => m === 'update')[1][0]
-      const mobUpdate = clientMob._builder._calls.find(([m]) => m === 'update')[1][0]
+      const webUpdate = clientWeb._builder._calls.find(([m]: any) => m === 'update')[1][0]
+      const mobUpdate = clientMob._builder._calls.find(([m]: any) => m === 'update')[1][0]
```
[the remark this answers is not in the document]
Comment (coelhotv) [reply packages/core/src/repositories/__tests__/createTreatmentPlanRepository.test.ts]: @gemini-code-assist fix applied in commit 08a3c9cd — can you validate?

File: packages/core/src/repositories/__tests__/createProfileRepository.test.ts [hunk 5]
```diff
@@ -73,10 +73,10 @@ describe('createProfileRepository — parity', () => {
       const out = await repo.getProfile()
 
       expect(client.from).toHaveBeenCalledWith('user_settings')
-      const calls = client._builder._calls.map((c) => c[0])
+      const calls = client._builder._calls.map((c: any) => c[0])
       expect(calls).toContain('select')
       expect(calls).toContain('maybeSingle')
-      expect(client._builder._calls.find((c) => c[0] === 'eq')[1]).toEqual(['user_id', FAKE_USER])
+      expect(client._builder._calls.find((c: any) => c[0] === 'eq')[1]).toEqual(['user_id', FAKE_USER])
```
[the remark this answers is not in the document]
Comment (coelhotv) [reply packages/core/src/repositories/__tests__/createProfileRepository.test.ts]: @gemini-code-assist fix applied in commit 08a3c9cd — can you validate?
